homeassistant/components/apple_tv/__init__.py

```diff
@@ -337,13 +329,36 @@ async def _connect(self, conf):
         self._dispatch_send(SIGNAL_CONNECTED, self.atv)
         self._address_updated(str(conf.address))
 
+        await self._async_setup_device_registry()
+
         self._connection_attempts = 0
         if self._connection_was_lost:
             _LOGGER.info(
                 'Connection was re-established to Apple TV "%s"', self.atv.service.name
             )
             self._connection_was_lost = False
 
+    async def _async_setup_device_registry(self):
+        attrs = {
+            "identifiers": {(DOMAIN, self.config_entry.unique_id)},
+            "manufacturer": "Apple",
+            "name": self.config_entry.data.get(CONF_NAME),
```

Comment:
Is the name optional in the config entry data?

Reply:
No, it's always present. I will fix that!